### ISSUE TITLE
Update library to reflect new changes to the API

### ISSUE THREAD

**Issue by antoineeripret** [joshcarty/google-searchconsole#44]
Hi @joshcarty, 

Google has just announced that its API was going to be updated to include more data: https://developers.google.com/search/blog/2021/10/search-analytics-discover-gnews. 

I've not yet created a PR for theses changes (I should have the time this week), but if you agree, we should apply the following modifications to your library: 

- Update the `searchType `parameter and use `type` instead to ensure support with future updates. 
- Update the documentation, stating that this parameter now supports news, video, image, web, discover (new) and googleNews (new) 
- Update the filter method documentation to state that it now supports REGEX (includingRegex & excludingRegex). 

Thanks :)

**Comment by antoineeripret** [comment]
Hi @joshcarty: 

I just realized that I made a mistake in my PR. When you query the API for `discover` or `googleNews`, `position` is not returned as a metric because it doesn't apply there. 

Therefore, the row 340 of query.py generated an error: 

**self.metrics = ['clicks', 'impressions', 'ctr', 'position']**

It should be fixable easily by adding a condition within the function, but I'm not sure how to access the `type` within it. Will something like this work? 

```
if self.type.lower() in ['discover', 'googlenews','google_news']:
   self.metrics = ['clicks', 'impressions', 'ctr']
else;
   self.metrics = ['clicks', 'impressions', 'ctr','position']
```

Sorry about that!
